app/code/Magento/Customer/Model/Plugin/CustomerNotification.php

```diff
@@ -99,4 +102,53 @@ public function beforeDispatch(AbstractAction $subject, RequestInterface $reques
             }
         }
     }
+
+    /**
+     * Return the shared request.
+     * If the request wasn't injected because of the backward compatible optional constructor dependency,
+     * create a new request instance.
+     *
+     * @return RequestInterface
+     */
+    private function getRequest(): RequestInterface
+    {
+        if (null === $this->request) {
+            $this->request = ObjectManager::getInstance()->get(RequestInterface::class);
+        }
+        return $this->request;
+    }
+
+    /**
+     * Because RequestInterface has no isPost method the check is requied before calling it.
+     *
+     * @return bool
+     */
+    private function isPostRequest(): bool
+    {
+        $request = $this->getRequest();
+
+        return method_exists($request, 'isPost') && $request->isPost();
```

Comment:
I think it’s better to check if request is instance of http request class

Reply:
Will check if it's `\Magento\Framework\App\HttpRequestInterface` instance.